src/main/java/org/magicdgs/http/jsr203/HttpFileSystem.java

```diff
@@ -116,13 +116,26 @@ public Path getPath(final String first, final String... more) {
         try {
             // handle the Path with the URI to separate Path query and fragment
             // in addition, it checks for errors in the encoding (e.g., null chars)
-            final URI uri = new URI(path);
-            return new HttpPath(this, uri.getPath(), uri.getQuery(), uri.getFragment());
+            return getPath(new URI(path));
```

Comment:
If the URI has a scheme and/or an authority, it should check if they are the same. Maybe this is for a different issue. The `getPath(URI)` package-protected method can live without doing any checking, because it is re-used by the provider...but it should specify the behaviour...